libs/langchain/langchain/memory/chat_message_histories/sql.py

```diff
@@ -49,44 +87,51 @@ def __init__(
         session_id: str,
         connection_string: str,
         table_name: str = "message_store",
+        session_id_field_name: str = "session_id",
+        custom_message_converter: Optional[BaseMessageConverter] = None,
     ):
         self.table_name = table_name
         self.connection_string = connection_string
         self.engine = create_engine(connection_string, echo=False)
+        self.session_id_field_name = session_id_field_name
+        self.converter = custom_message_converter or DefaultMessageConverter(table_name)
+        self.sql_model_class = self.converter.get_sql_model_class()
+        if not hasattr(self.sql_model_class, session_id_field_name):
+            raise ValueError("SQL model class must have session_id column")
         self._create_table_if_not_exists()
 
         self.session_id = session_id
         self.Session = sessionmaker(self.engine)
 
     def _create_table_if_not_exists(self) -> None:
-        DynamicBase = declarative_base()
-        self.Message = create_message_model(self.table_name, DynamicBase)
-        # Create all does the check for us in case the table exists.
-        DynamicBase.metadata.create_all(self.engine)
+        self.sql_model_class.metadata.create_all(self.engine)
 
     @property
     def messages(self) -> List[BaseMessage]:  # type: ignore
         """Retrieve all messages from db"""
         with self.Session() as session:
-            result = session.query(self.Message).where(
-                self.Message.session_id == self.session_id
+            result = session.query(self.sql_model_class).where(
+                getattr(self.sql_model_class, self.session_id_field_name)
+                == self.session_id
             )
-            items = [json.loads(record.message) for record in result]
-            messages = messages_from_dict(items)
+            messages = []
+            for record in result:
+                messages.append(self.converter.from_sql_model(record))
             return messages
 
     def add_message(self, message: BaseMessage) -> None:
         """Append the message to the record in db"""
         with self.Session() as session:
-            jsonstr = json.dumps(_message_to_dict(message))
-            session.add(self.Message(session_id=self.session_id, message=jsonstr))
+            session.add(self.converter.to_sql_model(message, self.session_id))
             session.commit()
 
     def clear(self) -> None:
         """Clear session memory from db"""
 
         with self.Session() as session:
-            session.query(self.Message).filter(
-                self.Message.session_id == self.session_id
+            session.commit()
```

Comment:
why this? (if actually necessary, lets leave comment)

Reply:
Removed the first commit invocation, the line was duplicated somehow when I changed the code around.